cromwell-backend/src/main/scala/cromwell/backend/provider/local/LocalBackend.scala

```diff
@@ -0,0 +1,348 @@
+package cromwell.backend.provider.local
+
+import java.nio.file.{Files, Path, Paths}
+import java.util.regex.Pattern
+
+import akka.actor.{ActorRef, Props}
+import akka.util.Timeout
+import better.files._
+import com.typesafe.scalalogging.StrictLogging
+import cromwell.backend.BackendActor
+import cromwell.backend.model._
+import cromwell.backend.provider.local.FileExtensions._
+import wdl4s.WdlExpression
+import wdl4s.types.{WdlFileType, WdlType}
+import wdl4s.values.{WdlFile, WdlSingleFile, WdlValue}
+
+import scala.annotation.tailrec
+import scala.collection.mutable.ArrayBuffer
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.sys.process._
+import scala.util.{Random, Try}
+
+object LocalBackend {
+  // Folders
+  val CromwellExecutionDir = "cromwell-executions"
+
+  // Files
+  val ReturnCodeFile = "rc"
+  val StdoutFile = "stdout"
+  val StderrFile = "stderr"
+  val ScriptFile = "script"
+
+  // Other
+  val DockerFlag = "docker"
+
+  def props(task: TaskDescriptor): Props = Props(new LocalBackend(task))
+}
+
+/**
+  * Executes a task in local computer through command line. It can be also executed in a Docker container.
+  * @param task Task descriptor.
+  */
+class LocalBackend(task: TaskDescriptor) extends BackendActor with StrictLogging {
+
+  import LocalBackend._
+
+  implicit val timeout = Timeout(10 seconds)
+  val subscriptions = ArrayBuffer[Subscription[ActorRef]]()
+
+  val workingDir = task.workingDir
+  val taskWorkingDir = task.name
+  val shardId = Random.nextInt(Integer.MAX_VALUE).toString;
+  val executionDir = Paths.get(CromwellExecutionDir, workingDir, taskWorkingDir, shardId).toAbsolutePath
+  val stdout = Paths.get(executionDir.toString, StdoutFile)
+  val stderr = Paths.get(executionDir.toString, StderrFile)
+  val script = Paths.get(executionDir.toString, ScriptFile)
+  val returnCode = Paths.get(executionDir.toString, ReturnCodeFile)
+  val stdoutWriter = stdout.untailed
+  val stderrTailed = stderr.tailed(100)
+  val argv = Seq("/bin/bash", script.toString)
+  val dockerImage = getDockerImage(task.runtimeAttributes)
+  var processAbortFunc: Option[() => Unit] = None
```

Comment:
I think the same but in this case is because of ProcessBuilder... I need to define a function to stop the process but I know the process info at the time of calling run function from ProcessBuilder.
If you know a better way please let me know... in the meantime I added private modifier.